### PR TITLE
Add no-module-prefix option

### PR DESCRIPTION
**Pull request by Herrie82**
There are other distributions besides SFOS that are using libconnman-qt via meta-qt5 Yocto layer (https://github.com/meta-qt5/meta-qt5/blob/master/recipes-connectivity/libconnman-qt/libconnman-qt5_git.bb) for example.

Currently meta-qt5 carriers a patch for removing the prefix (https://github.com/meta-qt5/meta-qt5/blob/master/recipes-connectivity/libconnman-qt/libconnman-qt5/0001-Don-t-use-MeeGo-as-prefix-in-order-to-make-this-a-co.patch), a cleaner solution is to upstream it by means of a config option which we're doing now. 

Can be enabled by building using "no-module-profix" config option. It should have no influence on the current operation for SFOS.

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>

### CODE REVIEW

File: plugin/files/noprefix/plugins.qmltypes [hunk 1]
```diff
@@ -0,0 +1,1017 @@
+import QtQuick.tooling 1.2
+
+// This file describes the plugin-supplied types contained in the library.
+// It is used for QML tooling purposes only.
+//
+// This file was auto-generated by:
+// 'qmlplugindump -nonrelocatable MeeGo.Connman 0.2'
```
Comment (rainemak) [inline plugin/files/noprefix/plugins.qmltypes]:
```suggestion
// 'qmlplugindump -nonrelocatable Connman 0.2'
```
A small typo.

**Review by rainemak — COMMENTED** [review]
The spec doesn't have option to pass "no-module-prefix" to the qmake and/or adjusting libdir of the connman-qt5-declarative. That can be done later if needed.

Added minor for the noprefix  plugins.qmltypes, other than that this looks good to me.

**Comment by Thaodan** [comment]
I wonder if it makes sense to remove the prefix altogether.

**Comment by rainemak** [comment]
> I wonder if it makes sense to remove the prefix altogether.

As of now, better to have the option for both.

**Comment by Herrie82** [comment]
I'm not a big fan of the duplicated files, but I didn't really find an option to manipulate the qmldir and plugins.qmltypes files to work around this using qmake

**Comment by rainemak** [comment]
> I'm not a big fan of the duplicated files, but I didn't really find an option to manipulate the qmldir and plugins.qmltypes files to work around this using qmake

If you'd like to get rid of duplicates, qmldir and plugins.types could have some find-and-replace variables and those variables could be replaced from the project file by with using [system command](https://doc.qt.io/qt-5/qmake-test-function-reference.html#system-command) e.g with sed. Downside being that using `sed` would be make this platform dependant but I doubt that this is much used in Windows.

**Comment by Herrie82** [comment]
> that

That sounds like something I'd like as well. Happy for suggestions. sed should work. I can do it in our build system (yocto/bitbake), but this wouldn't work with rpm and I don't know how to do it in qmake to be honest

**Comment by pvuorela** [comment]
On many qml modules we've done module renamings by adding backward compatibility symlinks etc, and if ok then at some point removed the compatibility support. E.g. nemo-qml-plugin-dbus still has both org.nemomobile.dbus and Nemo.DBus. 

Here it could be nice if there was a similar bath to gradually migrate to imports without MeeGo.

**Comment by rainemak** [comment]
> > that
> 
> That sounds like something I'd like as well. Happy for suggestions. sed should work. I can do it in our build system (yocto/bitbake), but this wouldn't work with rpm and I don't know how to do it in qmake to be honest

That doesn't need to affect the spec at all. I was thinking something like for plugin.pro:

```
no-module-prefix: {
    system("sed -i 's/##ReplaceMe##//' qmldir")
    MODULENAME = Connman
} else {
    system("sed -i 's/##ReplaceMe##/MeeGo./' qmldir")
    MODULENAME = MeeGo/Connman
    DEFINES += USE_MODULE_PREFIX
}
```

And qmldir would be like:
```
module ##ReplaceMe##Connman
plugin ConnmanQtDeclarative
typeinfo plugins.qmltypes
```

Feel free suggest something else for the ##ReplaceMe##. Then of course similar steps for the plugins.qmltypes. Take above sample code with a grant of salt as I only ran sed from the command line.

**Comment by rainemak** [comment]
Example of @pvuorela suggestion can be found from [nemo-qml-plugin-dbus](https://github.com/sailfishos/nemo-qml-plugin-dbus/blob/master/src/plugin/plugin.cpp#L42). 

To be noted that we do symlinking and seding of the qlmdir on the [spec](https://github.com/sailfishos/nemo-qml-plugin-dbus/blob/master/rpm/nemo-qml-plugin-dbus-qt5.spec#L56). As you are not working with rpm / spec, I can propose spec changes on top yours once this is landed.

**Comment by Herrie82** [comment]
@rainemak @pvuorela @Thaodan I pushed a commit to get rid of the duplicate files. Took me a while to get the syntax of sed right. Seems it didn't like the ## and the . after Meego. This works at my end now.

**Comment by rainemak** [comment]
> @rainemak @pvuorela @Thaodan I pushed a commit to get rid of the duplicate files. Took me a while to get the syntax of sed right. Seems it didn't like the ## and the . after Meego. This works at my end now.

Could you please squash your changes into one commit. Change itself looks good now.

**Comment by Herrie82** [comment]
> > @rainemak @pvuorela @Thaodan I pushed a commit to get rid of the duplicate files. Took me a while to get the syntax of sed right. Seems it didn't like the ## and the . after Meego. This works at my end now.
> 
> Could you please squash your changes into one commit. Change itself looks good now.

Done :) Thanks for pointers and feedback

**Review by rainemak — APPROVED** [review]

**Pull request merged**